webpack/scenes/Tasks/TaskActions.js

```diff
@@ -5,8 +5,16 @@ import { stopInterval, withInterval } from 'foremanReact/redux/middlewares/Inter
 import { foremanTasksApi } from '../../services/api';
 import { bulkSearchKey, pollTaskKey, taskFinishedToast } from './helpers';
 
-export const toastTaskFinished = task => async dispatch =>
+const finishedTasks = {};
+
+export const toastTaskFinished = task => async (dispatch) => {
```

Comment:
I'm wondering if this method care about whether we've already toasted about some task already, keeping in mind that this is probably the only area of the code affected by our problem.

Thinking more, maybe the issue isn't related to the task status at all. I have a theory it could be related to the SubscriptionsPage component being re-rendered while a manifest task is in progress.

See: https://github.com/Katello/katello/blob/master/webpack/scenes/Subscriptions/SubscriptionsPage.js#L45-L47
and
https://github.com/Katello/katello/blob/master/webpack/scenes/Subscriptions/SubscriptionActions.js#L109-L115

The problematic flow might be like this:

-> task polled
-> task has transitioned to finished
-> componentWillUpdate (task changed)
-> handleTask called
-> componentWillUpdate is called while handleTask is already executing, and now **handleTask is called again with the same completed task**

You might need to prevent this second `handleTask` from being called. Assuming this is the problem, is there a React way to handle it? Fixing the problem at this level seems like the way to go if it really is the problem, and it's probably pretty easy to write a test for.

Reply:
@jturel Thanks, you sent me on a journey :)

We were doing too much in `handleTask` anyway, so I split that into 2 actions.  I also moved all the logic to the component, which both feels better AND allows us to avoid using `getState()` in the thunk.  The new home for the logic is `componentDidUpdate` (I think that's what you meant above), which I chose so that we could use the `prevProps` for easy comparison.

Reply:
It looks awesome!!